tms/src/main/java/com/netcracker/edu/tms/ui/ProjectRestController.java

```diff
@@ -40,13 +46,26 @@
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
 
+        //this block must be before last return line, because in case of invalid addProject function block must not be executed
+        //start of the  block
```

Comment:
It is better to move logic related with mailing to service layer
and use method to wrap logic
`private void sendNotifications(List<User> addedUsers) {
... mailing logic
}`